pysus/online_data/Infodengue.py

```diff
@@ -0,0 +1,100 @@
+import json
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import pandas as pd
+from loguru import logger
+
+APP_DIR = Path(__file__).resolve(strict=True).parent.parent
+CID10 = {"dengue": "A90", "chikungunya": "A92.0", "zika": "A928"}
+
+with open(APP_DIR / "dataset/geocode_by_cities.json", "r") as f:
+    geocode_by_cities = json.load(f)
+
+
+def search_cities(city_name: str) -> Dict[str, int]:
+    """
+    Fetch geocode of the city name matching to the substring.
+
+    Parameters
+    ----------
+        city_name: Substring of city name
+    Returns
+    -------
+        dict: Dictionary with key and values 
+            with city name and IBGE codes of all municipalities in Brazil
+    """
+
+    result = []
+    for key in geocode_by_cities:
```

Comment:
This loop is not very efficient, Ideally you should create a simple inverted index with the name of the cities, as I explain [here](https://github.com/fccoelho/curso-IRI/blob/master/Pratica1.ipynb).
with it, you will only need 2 dictionary lookups to get the geocode

Reply:
and what about a fuzzy search?

Reply:
```
from fuzzywuzzy import fuzz

def search_string(substr: str, max_dist=3) -> Dict[str, int]:
    """
    Fetch geocode of the city name matching to the substring.

    Parameters
    ----------
        substr: Part of city name
    Returns
    -------
        dict: Dictionary with key and values
            with city name and IBGE codes of all municipalities in Brazil
    """

    return {
        key: geocode_by_cities[key]
        for key in geocode_by_cities
        if fuzz.token_set_ratio(substr, key) >= 67
        } # 67% matching percentage

####
%time search_string("s paulo")    
CPU times: user 78.6 ms, sys: 0 ns, total: 78.6 ms
Wall time: 78 ms

{'Altos': 67,
 'Engenheiro Paulo de Frontin': 83,
 'Frei Paulo': 83,
 'Monsenhor Paulo': 83,
 'Palmitos': 67,
 'Patos': 67,
 'Paudalho': 67,
 'Paulista': 67,
 'Paulo Afonso': 83,
 'Paulo Bento': 83,
 'Paulo Frontin': 83,
 'Paulo Jacinto': 83,
 'Paulo Lopes': 83,
 'Paulo Ramos': 83,
 'Paulo de Faria': 83,
 'São Julião': 67,
 'São Paulo': 93,
 'São Paulo das Missões': 83,
 'São Paulo de Olivença': 83,
 'São Paulo do Potengi': 83}

 ```

It looks much better than nltk in terms of fetch time and lines of code.
Thanks @xmnlab.

Reply:
I recommend using Python's [difflib.get_close_matches()](https://docs.python.org/3/library/difflib.html#difflib.get_close_matches) instead of Fuzzywuzzy. That way we avoid adding an extra external dependency. `difflib` in isn the Standard library.

Reply:
In that case:
```
from difflib import get_close_matches

def search_string(substr):       
    cities_lst = get_close_matches(substr, list(geocode_by_cities.keys()))
    cities_dict = {}
    
    for n in cities_lst:
        cities_dict[n] = geocode_by_cities[n]
        
    return cities_dict
    
###
%time search_string("s paulo")
CPU times: user 14.2 ms, sys: 0 ns, total: 14.2 ms
Wall time: 13.9 ms
{'São Paulo': 3550308}

 ```